internal/cmdrender/executor.go

```diff
@@ -77,24 +84,39 @@ func (e *Executor) Execute(ctx context.Context) error {
 		return err
 	}
 
-	pkgWriter := &kio.LocalPackageReadWriter{PackagePath: string(root.pkg.UniquePath)}
-
 	// format resources before writing
 	_, err = filters.FormatFilter{UseSchema: true}.Filter(resources)
 	if err != nil {
 		return err
 	}
 
-	err = pkgWriter.Write(resources)
-	if err != nil {
-		return fmt.Errorf("failed to save resources: %w", err)
+	if e.Output == nil {
+		pkgWriter := &kio.LocalPackageReadWriter{PackagePath: string(root.pkg.UniquePath)}
+		err = pkgWriter.Write(resources)
+		if err != nil {
+			return fmt.Errorf("failed to save resources: %w", err)
+		}
+	} else {
+		// write to the output instead of the pkg directory if e.Output is specified
+		writer := &kio.ByteWriter{
+			Writer:                e.Output,
+			KeepReaderAnnotations: true,
+			WrappingAPIVersion:    kio.ResourceListAPIVersion,
+			WrappingKind:          kio.ResourceListKind,
+		}
+		err = writer.Write(resources)
+		if err != nil {
+			return fmt.Errorf("failed to write resources: %w", err)
+		}
 	}
 
 	if err = pruneResources(hctx); err != nil {
 		return err
 	}
 
-	pr.Printf("Successfully executed %d function(s) in %d package(s).\n", hctx.executedFunctionCnt, len(hctx.pkgs))
+	if !disableCLIOutput {
+		pr.Printf("Successfully executed %d function(s) in %d package(s).\n", hctx.executedFunctionCnt, len(hctx.pkgs))
+	}
 
 	return e.saveFnResults(ctx, hctx.fnResults, false)
```

Comment:
We need to wire `disableCLIOutput` through `saveFnResults` as well.

Reply:
Why ? Results are orthogonal to where we write output resources.

Reply:
We also print a message saying where we saved the structured results. Pl. add test case for `fn eval` and `fn render` that involves out-of-place with `--results-dir` option (ensure we have atleast one `fn eval` call after the call that involves `--results-dir`, you should see the failure).

Reply:
updated and added test case.